File: progSys/processus/C/exo5.c
#include <time.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>

int main()
{
    srand(time(NULL));
    int time1=(rand()%10)+1;
    int time2=(rand()%10)+1;
    pid_t pid1=fork();
    if (pid1==0)
    {
        sleep(time1);
        printf("Je suis le fils 1 qui a duré %d secondes\n",time1);

    }
    else
    {
        pid_t pid2=fork();
        if (pid2==0)
        {
                sleep(time2);
                printf("Je suis le fils 2 qui a duré %d secondes\n",time2); 
        }
        else
        {
            if (time2>time1)
            {
                waitpid(pid2,NULL,0);
                waitpid(pid1,NULL,0);
                return 0;
            }
            else
            {
                waitpid(pid1,NULL,0);
                waitpid(pid2,NULL,0);
                return 0;
            }
        }
    }
}

File: progSys/processus/python/exo5.py
import os
import time
import random

random.seed()

time1 = random.randint(1, 10)
time2 = random.randint(1, 10)

pid1 = os.fork()

if pid1 == 0:
    time.sleep(time1)
    print(f"Je suis le fils 1 qui a duré {time1} secondes")
else:
    pid2 = os.fork()
    
    if pid2 == 0:

        time.sleep(time2)
        print(f"Je suis le fils 2 qui a duré {time2} secondes")
    else:
        if time2 > time1:
            os.waitpid(pid2, 0)
            os.waitpid(pid1, 0)
        else:
            os.waitpid(pid1, 0)
            os.waitpid(pid2, 0)